File: frontend/src/types/dispatch.ts
export interface AITag {
  id: number
  ai_tool: string
  workspace_id: number | null
  tags: Record<string, unknown>
  stats: Record<string, unknown>
  created_at: string
  updated_at: string
}

export interface AITagUpdate {
  tags?: Record<string, unknown>
  stats?: Record<string, unknown>
}

export interface DispatchRequest {
  task_id: number
  task_type?: string
  preferred_ai?: string
}

export interface DispatchResponse {
  id: number
  task_id: number
  ai_tool: string
  dispatch_reason: string | null
  risk_level: string | null
  execution_prompt: string | null
  success: boolean | null
  created_at: string
}

export interface DispatchRecord extends DispatchResponse {
  execution_result: string | null
  duration_seconds: number | null
}

File: backend/app/schemas/dispatch.py
from pydantic import BaseModel
from datetime import datetime


class AITagResponse(BaseModel):
    id: int
    ai_tool: str
    workspace_id: int | None
    tags: dict
    stats: dict
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class AITagUpdate(BaseModel):
    tags: dict | None = None
    stats: dict | None = None


class DispatchRequest(BaseModel):
    task_id: int
    task_type: str | None = None
    preferred_ai: str | None = None


class DispatchResponse(BaseModel):
    id: int
    task_id: int
    ai_tool: str
    dispatch_reason: str | None
    risk_level: str | None
    execution_prompt: str | None
    success: bool | None
    created_at: datetime

    class Config:
        from_attributes = True


class DispatchRecordResponse(DispatchResponse):
    execution_result: str | None = None
    duration_seconds: int | None = None
